tests/unit/lfp/lfp.cpp

```diff
@@ -0,0 +1,104 @@
+#define BOOST_TEST_MODULE LFPTest
+#define BOOST_TEST_MAIN
+
+#include <iostream>
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/test_case_template.hpp>
+
+#include "coreneuron/io/lfp.hpp"
+#include "coreneuron/mpi/nrnmpi.h"
+
+template <typename F>
+double integral(F f, double a, double b, int n) {
+    double step = (b - a) / n;  // width of each small rectangle
+    double area = 0.0;          // signed area
+    for (int i = 0; i < n; i++) {
+        area += f(a + (i + 0.5) * step) * step;  // sum up each small rectangle
+    }
+    return area;
+}
+
+std::array<double, 3> operator*(const double x, const std::array<double, 3>& y) {
+    return {x * y[0], x * y[1], x * y[2]};
+}
+
+std::array<double, 3> operator+(const std::array<double, 3>& x, const std::array<double, 3>& y) {
```

Comment:
following the previous comment. If you implement the multiplication by scalar the axp function can be eliminated

Reply:
I removed the function entirely, relying instead, for ease, on paxpy.